File: Frontend/src/app/entity/ReqRes.ts
import {UserRole} from "./UserRole";
import {User} from "./UserEntity";
import { NumberValueAccessor } from "@angular/forms";

export class ReqRes {
  id: number
  statusCode: number;
  error: string;
  message: string;
  token: string;
  refreshToken: string;
  expirationTime: string;
  vorname: string;
  nachname: string;
  benutzername: string;
  email: string;
  passwort: string;
  roles: UserRole.USER;
  user: string;

  constructor(
    id: number,
    statusCode: number,
    error: string,
    message: string,
    token: string,
    refreshToken: string,
    expirationTime: string,
    vorname: string,
    nachname: string,
    benutzername: string,
    email: string,
    passwort: string,
    roles: UserRole.USER,
    user: string
  ) {
    this.id = id
    this.statusCode = statusCode;
    this.error = error;
    this.message = message;
    this.token = token;
    this.refreshToken = refreshToken;
    this.expirationTime = expirationTime;
    this.vorname = vorname;
    this.nachname = nachname;
    this.benutzername = benutzername;
    this.email = email;
    this.passwort = passwort;
    this.roles = roles;
    this.user = user;
  }
}

File: Backend/src/main/java/com/example/backend/dto/ReqRes.java
package com.example.backend.dto;

import com.example.backend.entity.User;
import com.example.backend.entity.UserRole;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ReqRes {

    private int statusCode;
    private String error;
    private String message;
    private String token;
    private String refreshToken;
    private String expirationTime;
    private String vorname;
    private String nachname;
    private String benutzername;
    private String email;
    private String passwort;
    private List<UserRole> roles;
    private User user;
}